src/nvim/eval.c

```diff
@@ -11780,7 +11781,7 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
       TV_DICT_ITER(item->di_tv.vval.v_dict, var, {
         const char *str = tv_get_string(&var->di_tv);
         if (str) {
-          size_t len = STRLEN(var->di_key) + strlen(str) + strlen("=");
+          size_t len = STRLEN(var->di_key) + strlen(str) + strlen("=") + 1;
```

Comment:
`xmallocz()` adjusts the allocation length to account for the NUL.  It looks like you're including that in the calculation here so that `snprintf()` has an accurate length to work with, in which case the `xmallocz()` should be changed to `xmalloc()`.